ravenframework/TSA/Transformers/FilterBankDWT.py

```diff
@@ -0,0 +1,251 @@
+      \item \textbf{coif family}: coif1, coif2, coif3, coif4, coif5, coif6, coif7, coif8,
+        coif9, coif10, coif11, coif12, coif13, coif14, coif15, coif16, coif17
+      \item \textbf{bior family}: bior1.1, bior1.3, bior1.5, bior2.2, bior2.4, bior2.6,
+        bior2.8, bior3.1, bior3.3, bior3.5, bior3.7, bior3.9, bior4.4, bior5.5,
+        bior6.8
+      \item \textbf{rbio family}: rbio1.1, rbio1.3, rbio1.5, rbio2.2, rbio2.4, rbio2.6,
+        rbio2.8, rbio3.1, rbio3.3, rbio3.5, rbio3.7, rbio3.9, rbio4.4, rbio5.5,
+        rbio6.8
+      \item \textbf{dmey family}: dmey
+      \item \textbf{gaus family}: gaus1, gaus2, gaus3, gaus4, gaus5, gaus6, gaus7, gaus8
+      \item \textbf{mexh family}: mexh
+      \item \textbf{morl family}: morl
+      \item \textbf{cgau family}: cgau1, cgau2, cgau3, cgau4, cgau5, cgau6, cgau7, cgau8
+      \item \textbf{shan family}: shan
+      \item \textbf{fbsp family}: fbsp
+      \item \textbf{cmor family}: cmor
+    \end{itemize}"""))
+    specs.addSub(InputData.parameterInputFactory('levels', contentType=InputTypes.IntegerType,
+              descr=r"""the number of wavelet decomposition levels for our signal. If level is 0,
+                    it doesn't """))
+    return specs
+
+  def handleInput(self, spec):
+    """
+      Reads user inputs into this object.
+      @ In, spec, InputData.InputParams, input specifications
+      @ Out, settings, dict, initialization settings for this algorithm
+    """
+    settings = super().handleInput(spec)
+    settings['family'] = spec.findFirst('family').value
+    settings['levels'] = spec.findFirst('levels').value
+
+    self._levels = settings['levels'] - 1
+    return settings
+
+  def fit(self, signal, pivot, targets, settings, trainedParams=None):
+    """
+      This function utilizes the Discrete Wavelet Transform to
+      characterize a time-dependent series of data.
+
+      @ In, signal, np.ndarray, time series with dims [time, target]
+      @ In, pivot, np.1darray, time-like parameter values
+      @ In, targets, list(str), names of targets in same order as signal
+      @ In, settings, dict, additional settings specific to this algorithm
+      @ In, trainedParams, dict, running dict of trained algorithm params
+      @ Out, params, dict, characteristic parameters
+    """
+    # TODO extend to continuous wavelet transform
+
+    ## The pivot input parameter isn't used explicity in the
+    ## transformation as it assumed/required that each element in the
+    ## time-dependent series is independent, uniquely indexed and
+    ## sorted in time.
+    family = settings['family']
+    params = {target: {'results': {}} for target in targets}
+
+    # determine maximum decomposition level
+    max_level = pywt.dwt_max_level(len(pivot), family)
+    if self._levels>max_level:
+      print(f"Number of levels requested is larger than maximum DWT decomposition level, switching to maximum allowed: {max_level}")
```

Comment:
please use message handler to raise messages.

Reply:
it looks like the base classes for the TSA algorithms do not inherit Message Handler at the moment. this may be a subject of future discussion?

Reply:
This is ok to me. I would suggest you to talk to Paul to see if it makes sense to inherit message handler for TSA classes. @PaulTalbot-INL

Reply:
We've revisited this many times; I think we need RAVEN to move to using a logger, then it would make sense, rather than pushing TSA to inherit from additional classes.